.golangci.yml

```diff
@@ -3,15 +3,30 @@
 # Copyright (c) 2023 HashiCorp, Inc.
 # SPDX-License-Identifier: MPL-2.0
 
-run:
-  timeout: 30m
-
-issues:
-  exclude-files:
-    # We have a few patterns that are excluded from linting completely because
-    # they contain effectively-frozen code that we're preserving for backward
-    # compatibility, where changes would be risky and that risk isn't warranted
-    # since we don't expect to be doing any significant maintenence on these.
-    - "^internal/ipaddr/"
-    - "^internal/legacy/"
-    - "^internal/states/statefile/version\\d+_upgrade\\.go$"
+version: "2"
+linters:
```

Comment:
Suggestion:
```diff
+  settings:
+    staticcheck:
+      checks: ["all", "-QF1008", "-ST1003", "-ST1005", "-ST1012", "-ST1016"]
```

That mirrors our current configuration, with two small exceptions we could just fix in this PR.

Reply:
We should also include a comment to address those once the work to remove `--new-from-rev` is done.

Reply:
Also considering "-QF1001": "could apply De Morgan's law"